Src/FluentAssertions/Collections/GenericCollectionAssertions.cs

```diff
@@ -2917,7 +2916,7 @@ protected static IEnumerable<TExpectation> RepeatAsManyAs<TExpectation>(TExpecta
                 .ClearExpectation();
         }
 
-        protected void AssertSubjectEquality<TActual, TExpectation>(IEnumerable<TExpectation> expectation, Func<TActual, TExpectation, bool> equalityComparison,
+        protected void AssertSubjectEquality<TExpectation>(IEnumerable<TExpectation> expectation, Func<T, TExpectation, bool> equalityComparison,
```

Comment:
The generics was not needed as `TActual` is now equal to `T` and `Subject` is already of type `IEnumerable<T>`